### PR TITLE
My Home: Tweak styles for experimental layout

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request

Tweak some styles and add alternative versions of some cards for better integration in the experimental Home layout, so the styling is more aligned with the mockups shared in pbAPfg-g0-p2.

Before | After
--- | ---
image | image
image | image

- New compact view for the quick links.
- Educational content is stacked (one card below the other) in a shared section.
- Font sizes and spacing have been adjusted.
- Display the "Connect accounts" task if there are no accounts connected.

If this PR feels too big, I'm happy to split work in different PRs.

#### Testing instructions

- Visit My Home without enabling the experimental layout: `/home/:site`.
- Make sure the layout looks like in production (there should be minimal sizing/spacing changes).
- Visit My Home enabling the experimental layout: `/home/:site?flags=home/experimental-layout`.
- Make sure the layout looks good and there are no styling glitches. 
- Try different sites (prelaunch, setup, established, WP for Teams) and make sure the layout is still consistent.
- Apply D42141-code and sandbox the API.
- Visit My Home enabling the experimental layout for an established site without any Publicize connection.
- Make sure there is a task in the primary location for connecting accounts.

### CODE REVIEW

File: client/my-sites/customer-home/cards/actions/quick-links-compact/index.jsx [hunk 1]
```diff
@@ -0,0 +1,309 @@
+/**
```
Comment (mmtr) [inline client/my-sites/customer-home/cards/actions/quick-links-compact/index.jsx:1]: This is basically copied from the [wide version of the quick links card](https://github.com/Automattic/wp-calypso/blob/5fb0b042c067a32feaa9a4f00d8b7e90969cc95b/client/my-sites/customer-home/cards/actions/quick-links/index.jsx) but always returning a `FoldableCard`.

File: client/my-sites/customer-home/cards/actions/wp-for-teams-quick-links-compact/index.jsx [hunk 2]
```diff
@@ -0,0 +1,226 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
```
Comment (mmtr) [inline client/my-sites/customer-home/cards/actions/wp-for-teams-quick-links-compact/index.jsx:1]: This one is also mostly a copy-paste from [its wide counterpart version](https://github.com/Automattic/wp-calypso/blob/5fb0b042c067a32feaa9a4f00d8b7e90969cc95b/client/my-sites/customer-home/cards/actions/wp-for-teams-quick-links/index.jsx) just to ensure we use `FoldableCard` component.

File: client/my-sites/customer-home/cards/education/free-photo-library/index.jsx [hunk 3]
```diff
@@ -3,76 +3,42 @@
...
-					<img
-						className="free-photo-library__demonstration-image"
-						src={ freePhotoLibraryVideoPrompt }
-						alt={ translate( 'Free Photo Library demonstration' ) }
-					/>
-				</button>
+		<EducationalContent
+			title={ translate( 'The WordPress.com free photo library' ) }
+			description={ translate(
+				'Our free photo library integrates your site with over 40,000 beautiful copyright-free photos to create stunning designs.'
```
Comment (a8ci18n) [inline client/my-sites/customer-home/cards/education/free-photo-library/index.jsx:26]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'The WordPress.com Free Photo Library integrates your site with beautiful copyright-free photos to create stunning designs.' )` *ES Score: 8*

File: client/my-sites/customer-home/cards/actions/quick-links-compact/index.jsx (hunk 1 above)
Comment (gwwar) [reply client/my-sites/customer-home/cards/actions/quick-links-compact/index.jsx:1]: Noice, totally fine to copy paste here. We can always delete the other card later if we don't end up diverging/need it.

File: client/my-sites/customer-home/cards/education/free-photo-library/index.jsx [hunk 4]
```diff
@@ -3,76 +3,42 @@
...
+					supportLink={ localizeUrl( 'https://wordpress.com/support/free-photo-library/' ) }
+					showIcon={ false }
+					text={ translate( 'Learn more' ) }
+					tracksEvent="calypso_customer_home_free_photo_library_video_support_page_view"
+					statsGroup="calypso_customer_home"
+					statsName="view_free_photo_library_video"
+				/>
+			}
+			illustration={ freePhotoLibraryVideoPrompt }
+		/>
```
Comment (gwwar) [inline client/my-sites/customer-home/cards/education/free-photo-library/index.jsx:40]: Thanks for tidying 🧹

File: client/my-sites/customer-home/cards/actions/wp-for-teams-quick-links-compact/index.jsx [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+	writePostAction,
+	addPageAction,
+	manageCommentsAction,
+	trackEditMenusAction,
+	trackCustomizeThemeAction,
+} ) => {
+	const translate = useTranslate();
+
+	const quickLinks = (
+		/* eslint-disable wpcalypso/jsx-classname-namespace */
```
Comment (gwwar) [inline client/my-sites/customer-home/cards/actions/wp-for-teams-quick-links-compact/index.jsx:54]: Should we maybe follow the className naming here so folks can search for the component a bit more easily?

**Comment by gwwar** [comment]
> If this PR feels too big, I'm happy to split work in different PRs.

I think this is a good chuck provided we test for regressions in the existing layout 👍 

Code looks reasonable and I'm liking the updates!

I'll of course defer to @Automattic/dotcom-manage-design  but I left a few annotation notes here for some minor styling pieces/behavior that stuck out.

Screen Shot 2020-04-17 at 9 37 12 AM

Here's a screen to record the WIP with quick links expanded:
Screen Shot 2020-04-17 at 9 38 02 AM

**Comment by gwwar** [comment]
@Aurorum's connection task is in master, so we could potentially hook this up in the experimental view as well to help tweak styles.

**Comment by kwight** [comment]
We'll need to watch this doesn't clobber any upcoming heading improvements in #40997.

File: client/my-sites/customer-home/cards/actions/wp-for-teams-quick-links-compact/index.jsx (hunk 5 above)
Comment (mmtr) [reply client/my-sites/customer-home/cards/actions/wp-for-teams-quick-links-compact/index.jsx:54]: Yeah, sure. Main reason why didn't want to follow the classname namespace pattern was to reuse the regular quick links styles, but we can actually use 2 classes here 🙂

Comment (mmtr) [reply client/my-sites/customer-home/cards/actions/wp-for-teams-quick-links-compact/index.jsx]: Done in a62ba00

**Comment by mmtr** [comment]
> Card headings too bold?

I had the same feeling at first but I'm now getting used to it (and even liking it 🙂). In any case, I just tried to replicate the @danhauk's mockups from pbAPfg-g0-p2. In those mockups headers are bolder than the rest of the content, and AFAIK the only `font-weight` value we can use for that is 700. cc @sixhours 

> Make toggle behavior sticky like sidebar?

Yup, was planning to do it in a follow-up so this PR doesn't get bigger.

> Extra spacing between links.

Covered this on cd1b812 and also added icons to match mockups from pbAPfg-g0-p2.

Screen Shot 2020-04-20 at 12 37 37

> @Aurorum's connection task is in master, so we could potentially hook this up in the experimental view as well to help tweak styles.

Also covered this in 25012f3 and D42141-code.

File: client/my-sites/customer-home/cards/education/mastering-gutenberg/index.jsx [hunk 6]
```diff
@@ -3,60 +3,42 @@
...
+					url: localizeUrl( 'https://wordpress.com/support/wordpress-editor/#blocks' ),
+					text: translate( 'Adding and moving blocks' ),
+					icon: 'add-outline',
+					tracksEvent: 'calypso_customer_home_customizing_with_blocks_support_page_view',
+					statsName: 'view_customizing_with_blocks_video',
+				},
+				{
+					postId: 147594,
+					url: localizeUrl( 'https://wordpress.com/support/wordpress-editor/#configuring-a-block' ),
+					text: translate( 'Adjust settings of blocks' ),
```
Comment (a8ci18n) [inline client/my-sites/customer-home/cards/education/mastering-gutenberg/index.jsx:34]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Adjusting settings of blocks' )` *ES Score: 9*

**Comment by kwight** [comment]
I'm not sure about the icons – the two displayed look like they're from two different icon sets when close together like that, and it's kinda visually confusing when the card below doesn't have an icon at all ¯\_(ツ)_/¯ Everything else looks great, but I think there's a move to a `600` [font weight for bold](https://github.com/Automattic/wp-calypso/issues/41234#issuecomment-616815939).

Screen Shot 2020-04-20 at 3 12 09 PM

**Comment by Aurorum** [comment]
For what it's worth, there was previous discussion against those icons in #39617 

> Because those icons are for affordances (meaning they represent specific actions or functionality in the UI), they shouldn't be used as bullet points or other styling that starts to give them multiple meanings. This messes with users who are trying to learn something new

File: client/my-sites/customer-home/cards/education/mastering-gutenberg/index.jsx [hunk 7]
```diff
@@ -3,60 +3,42 @@
...
-				/>
-			</div>
-			{ ! isMobile() && (
-				<div className="mastering-gutenberg__illustration">
-					<img src="/calypso/images/illustrations/gutenberg-mini.svg" alt="" />
-				</div>
+		<EducationalContent
+			title={ translate( 'Master the Block Editor' ) }
+			description={ translate(
+				'Learn how to create stunning post and page layouts with our short video guides.'
```
Comment (a8ci18n) [inline client/my-sites/customer-home/cards/education/mastering-gutenberg/index.jsx:20]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Learn how to create stunning post and page layouts through our video guides.' )` *ES Score: 7*

**Comment by danhauk** [comment]
Thanks for this @mmtr, it's looking great. I just committed a few small tweaks to the styles to clean it up just a bit.

* Adjusted some padding, line-height, and font-sizes.
* Added the bottom border on the heading to provide some visual separation between the page heading and page content when there's no active next task.
* I kept the icons on the Gutenberg card to make those links more easily scannable as two separate links. But I did update them to `video` to provide some affordance that these are videos to watch.

Screenshot_2020-04-21 My Home ‹ Dan Hauk — WordPress com(1)

Screenshot_2020-04-21 My Home ‹ Site Titles — WordPress com

**Comment by mmtr** [comment]
Thanks @danhauk!

Now that https://github.com/Automattic/wp-calypso/pull/41330 has landed, I could set the card header's font weight to 600, so I think this should be ready for a final review.

**Comment by sixhours** [comment]
This tests well for me! I noticed some minor, non-blocking visual stuff:

Spacing in the setup list header looks off:

Screen Shot 2020-04-22 at 11 06 38 AM

The movie icons are squished:

Screen Shot 2020-04-22 at 11 07 37 AM

I updated the graphic in the main task area to have a max-width of 320px instead of 280px; I think it aligns better with the grid and evens out the height of those two columns. This borders on subjective, though, so @danhauk can make the final call there:

Before

Screen Shot 2020-04-22 at 11 18 46 AM

After

Screen Shot 2020-04-22 at 11 18 16 AM

File: client/my-sites/customer-home/cards/actions/quick-links-compact/index.jsx [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+				target="_blank"
+				label={ translate( 'Create a logo with Looka' ) }
+				external
+				iconSrc={ logoIcon }
+			/>
+		</div>
+	);
+
+	return (
+		<FoldableCard className="quick-links-compact" header={ translate( 'Quick links' ) }>
```
Comment (kwight) [inline client/my-sites/customer-home/cards/actions/quick-links-compact/index.jsx:148]: Hm, why is this `FoldableCard` not foldable?..

**Comment by mmtr** [comment]
> Spacing in the setup list header looks off

Yeah, but we shouldn't worry about this. We won't render this setup list in the new layout since we'll implement a redesigned version of the current card. Until then, it's normal we have these small glitches on the experimental layout.

> The movie icons are squished
> max-width of 320px instead of 280px

I think we can iterate on this in follow-ups. @danhauk feel free to open issues/PRs if you think this needs to be addressed. 

Said that, and given there isn't any real blocker design-wise, I'm going to merge this so we can continue improving the experimental layout and adding more cards to it.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3361888

Thank you @mmtr for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3361888).

**Pull request merged**